File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialog/DialogSaveClip.java
package com.iamkatrechko.clipboardmanager.view.dialog;

import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.iamkatrechko.clipboardmanager.R;

/**
 * Диалог сохранения заметки перед выходом
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
public class DialogSaveClip extends DialogFragment {

    /**
     * Возвращает новый экземпляр диалога
     * @return новый экземпляр диалога
     */
    public static DialogSaveClip newInstance() {
        return new DialogSaveClip();
    }

    /**
     * Возвращает данные обратно во фрагмент
     * @param save флаг сохранения заметки
     */
    private void sendResult(boolean save) {
        if (getTargetFragment() == null) {
            return;
        }
        Intent a = new Intent();
        a.putExtra("save", save);
        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, a);
    }

    @NonNull
    public Dialog onCreateDialog(Bundle bundle) {

        return new AlertDialog.Builder(getActivity())
                .setTitle(R.string.question_cancel)
                .setMessage(R.string.question_exit_without_save)
                .setPositiveButton(R.string.yes,
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog,
                                                int id) {
                                sendResult(false);
                            }
                        })
                .setNegativeButton(R.string.cancel, null).create();
    }
}

File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialog/DialogSaveClip.kt
package com.iamkatrechko.clipboardmanager.view.dialog

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import com.iamkatrechko.clipboardmanager.R
import com.iamkatrechko.clipboardmanager.view.extension.onActivityResultOk

/**
 * Диалог сохранения заметки перед выходом
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
class DialogSaveClip : DialogFragment() {

    /**
     * Возвращает данные обратно во фрагмент
     * @param save флаг сохранения заметки
     */
    private fun sendResult(save: Boolean) {
        onActivityResultOk {
            putExtra(KEY_IS_SAVE, save)
        }
    }

    override fun onCreateDialog(bundle: Bundle?): Dialog {
        return AlertDialog.Builder(activity!!)
                .setTitle(R.string.question_save)
                .setMessage(R.string.question_save_on_exit)
                .setPositiveButton(R.string.yes) { _, _ -> sendResult(true) }
                .setNegativeButton(R.string.no) { _, _ -> sendResult(false) }
                .setNeutralButton(R.string.cancel, null)
                .create()
    }

    companion object {

        const val KEY_IS_SAVE = "KEY_IS_SAVE"

        @JvmStatic
        fun newInstance(): DialogSaveClip =
                DialogSaveClip()
    }
}
